libs/apigateway-lambda-proxy/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,376 @@
+import * as pulumi from '@pulumi/pulumi'
+import * as aws from '@pulumi/aws'
+
+const namespace = 'AWS/ApiGateway'
+
+/**
+ * Dimensions to identify an HTTP API in CloudWatch.
+ *
+ * The main difference is REST APIs are identified by name, while HTTP APIs are identified by id.
+ */
+export type HttpGateway = {
+    /**
+     * The API Gateway ID to monitor. Only valid for HTTP endpoints.
+     */
+    id: pulumi.Input<string>
+
+    /**
+     * The API name to monitor.
+     *
+     * For REST endpoints, this must be the name of the API created in AWS.
+     * For HTTP endpoints, this can be anything (it's only used in alerts).
+     */
+    name: pulumi.Input<string> | undefined
+
+    /**
+     * The stage to monitor
+     */
+    stage?: pulumi.Input<string>
+}
+
+/**
+ * Dimensions to identify an API in CloudWatch.
+ */
+export type ApiGateway = HttpGateway
+
+/**
+ * Provides an opinionated set of recommended alarms for an API Gateway stage.
+ *
+ * Raise alarms for the following:
+ *  - 5xx error rate exceeded (default: 2%)
+ *  - 4xx error rate exceeded (default: 50%)
+ *  - integration latency too high (anomaly detection)
+ */
+export class RecommendedAlarms extends pulumi.ComponentResource {
+    constructor(
+        name: string,
+        args: {
+            /**
+             * Alerts will be sent to this SNS topic
+             */
+            snsTopicArn: pulumi.Input<string>
+            /**
+             * Set thresholds for the alarms
+             */
+            thresholds?: {
+                errorRate5xxPercent?: pulumi.Input<number>
+                errorRate4xxPercent?: pulumi.Input<number>
+                integrationLatencyStdDeviations?: pulumi.Input<number>
+            }
+            /**
+             * The API Gateway to monitor
+             */
+            apiGateway: ApiGateway
+            /**
+             * a callback function that returns tags for
+             * each resource
+             */
+            getTags: (
+                name: string,
+            ) => {
+                [key: string]: pulumi.Input<string>
+            }
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('wanews:apigateway/RecommendedAlarms', name, {}, opts)
```

Comment:
Why all lowercase except RecommendedAlarms?

Reply:
https://www.pulumi.com/docs/intro/concepts/resources/#authoring-a-new-component-resource

```
pkg:index:MyComponent
```

Should really be `apigateway-lamba-proxy:metric-alarms/RecommendedAlarms`, but that's pretty long, making for some funky console output, so I shortened it.